torchgen/model.py

```diff
@@ -1328,6 +1324,8 @@ def is_list_like(self) -> Optional["ListType"]:
         "Stream",
         "SymInt",
         "ConstQuantizerPtr",  # TODO: rename
+        "ScalarVal",  # An actual aten Scalar value, no special handling
```

Comment:
@antoniojkim what is the reason we had to make a separate 'ScalarVal' when we already had 'Scalar'?

Reply:
`Scalar` is treated specially and is replaced with a `Value` object and handled the same as a `Tensor`. However, for the `Scalar` IR node, we need an actual `c10::Scalar` value that isn't handled specially.

Reply:
`foo(Tensor a, Scalar b) -> Tensor` would emit something like
```
foo(Value a, Value b)
```
whereas the intention for `foo(Tensor a, ScalarVal b) -> Tensor` would be to emit something like
```
foo(Value a, Scalar b)
```
This isn't technically just MLIR specific. This is to codegen the `Scalar` IR node which requires specifically to take in a `Scalar` value.

Reply:
Got it. One way to handle this without having to reach into and tweak the codegen data model would be with some extra LT-specific metadata for the operator:
```
func: foo(Tensor a, Scalar b) -> Tensor
- treat_scalars_as_constants: True
```

Reply:
(note we wouldn't want to add this metadata to the `NativeFunction` data class, since that's general codegen. We'd want to plumb this somewhere so it only shows up in LT codegen. Maybe the `LazyIrSchema`?